src/renderer/kube-object.store.ts

```diff
@@ -24,6 +26,36 @@ export abstract class KubeObjectStore<T extends KubeObject = any> extends ItemSt
     this.bindWatchEventsUpdater();
   }
 
+  // TODO: detach / remove circular dependency
+  @observable.ref private namespaceStore: NamespaceStore;
+
+  private async resolveNamespaceStore(): Promise<NamespaceStore> {
+    const { namespaceStore } = await import("./components/+namespaces/namespace.store");
+
+    await namespaceStore.whenReady;
+    this.namespaceStore = namespaceStore;
+
+    return namespaceStore;
+  }
+
+  // TODO: detach / inject dependency as in kube-watch-api
+  @observable.ref private cluster: Cluster;
+
+  private async resolveCluster(): Promise<Cluster> {
+    const { getHostedCluster } = await import("../common/cluster-store");
+
+    this.cluster = getHostedCluster();
+
+    return this.cluster;
+  }
+
+  // TODO: figure out how to transparently replace with this.items
```

Comment:
Not sure if it's still needed. Probably not since default `replace=false` for `loadAll()`?

Reply:
Anyway, this is currently not used anywhere `computed` field.